code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1054,3 +1056,248 @@ var/list/coefflist = list()
 /mob/living/human/proc/make_adrenaline(amount)
 	if(stat == CONSCIOUS)
 		reagents.add_reagent("adrenaline", amount)
+
+/mob/living/human/proc/handle_ui_visibility(mob/living/user)
+	var/mob/living/human/H = user
+	if (H.client)
+		if (H.using_look())
+			for (var/obj/O in H.client.screen)
+				if (O.invisibility)
+					continue
+				if (istype(O, /obj/screen/movable/action_button))
+					var/obj/screen/movable/action_button/A = O
+					if (A.name == "Look into Distance ([src])" || (A.owner && istype(A.owner, /datum/action/toggle_scope)))
+						continue
+				O.invisibility = 100
+				O.scoped_invisible = TRUE
+		else
+			for (var/obj/O in H.client.screen)
+				if (O.scoped_invisible)
+					O.invisibility = FALSE
+
+/mob/living/human/proc/can_look(mob/living/user)//Largely copied from zoom.dm
+	var/mob/living/human/H = user
+
+	if(user.stat || !ishuman(user))
+		user << "<span class = 'warning'>You are unable to look into the distance right now.</span>"
+		return FALSE
+	if (H.wear_mask && istype(H.wear_mask, /obj/item/clothing/mask))
+		var/obj/item/clothing/mask/currmask = H.wear_mask
+		if (currmask.blocks_scope)
+			user << "You can't see any farther while wearing \the [currmask]!"
+			return FALSE
+	else if (global_hud.darkMask[1] in user.client.screen)
+		user << "Your visor gets in the way of seeing further."
+		return FALSE
+	else if (user.client.pixel_x || user.client.pixel_y) //Resets people's view if they're looking into the distance and try to look through something else
+		user.client.pixel_x = 0
+		user.client.pixel_y = 0
+		return FALSE
+	else
+		var/obj/item/organ/eyes/E = H.internal_organs_by_name["eyes"]
+		if (E.is_bruised() || E.is_broken() || H.eye_blind > 0)
+			if (!silent) user << "<span class = 'danger'>Your eyes are injured! You can't use \the [src].</span>"
```

Comment:
"you cant use the src" is gonna refer to the human, should be "you cant look in teh distance" or something